java/core/src/test/org/apache/orc/impl/TestConvertTreeReaderFactory.java

```diff
@@ -65,30 +143,163 @@ public void testArraySizeBiggerThan1024AndConvertToDecimal() throws Exception {
     assertEquals(batch.cols.length, 1);
     assertTrue(batch.cols[0] instanceof ListColumnVector);
     assertEquals(((ListColumnVector) batch.cols[0]).child.getClass(), expectedColumnType);
-    return (TExpectedColumn) ((ListColumnVector) batch.cols[0]).child;
+    return (TExpectedColumnVector) ((ListColumnVector) batch.cols[0]).child;
+  }
+
+  public void testConvertToDecimal() throws Exception {
+    Decimal64ColumnVector columnVector =
+        readORCFileWithLargeArray("decimal(6,1)", Decimal64ColumnVector.class);
+    assertEquals(LARGE_BATCH_SIZE, columnVector.vector.length);
+  }
+
+  public void testConvertToVarchar() throws Exception {
+    BytesColumnVector columnVector = readORCFileWithLargeArray("varchar(10)", BytesColumnVector.class);
+    assertEquals(LARGE_BATCH_SIZE, columnVector.vector.length);
+  }
+
+  public void testConvertToDouble() throws Exception {
+    DoubleColumnVector columnVector = readORCFileWithLargeArray("double", DoubleColumnVector.class);
+    assertEquals(LARGE_BATCH_SIZE, columnVector.vector.length);
+  }
+
+  public void testConvertToInteger() throws Exception {
+    LongColumnVector columnVector = readORCFileWithLargeArray("int", LongColumnVector.class);
+    assertEquals(LARGE_BATCH_SIZE, columnVector.vector.length);
+  }
+
+  public void testConvertToTimestamp() throws Exception {
+    TimestampColumnVector columnVector =
+        readORCFileWithLargeArray("timestamp", TimestampColumnVector.class);
+    assertEquals(LARGE_BATCH_SIZE, columnVector.time.length);
+  }
+
+  public void testConvertToDate() throws Exception {
+    DateColumnVector columnVector = readORCFileWithLargeArray("date", DateColumnVector.class);
+    assertEquals(LARGE_BATCH_SIZE, columnVector.vector.length);
   }
 
   @Test
-  public void testArraySizeBiggerThan1024AndConvertToVarchar() throws Exception {
-    BytesColumnVector columnVector = testArraySizeBiggerThan1024("varchar(10)", BytesColumnVector.class);
-    assertEquals(columnVector.vector.length, 1025);
+  public void testDecimalArrayBiggerThanDefault() throws Exception {
+    String typeStr = "decimal(6,1)";
+    Class typeClass = DecimalColumnVector.class;
+
+    TypeDescription schema = TypeDescription.fromString("struct<col1:array<" + typeStr + ">>");
+    createORCFileWithLargeArray(schema, typeClass, typeClass.equals(Decimal64ColumnVector.class));
+    // Test all possible conversions
```

Comment:
Does this include all? Could you add the reference comment pointing to the ORC?

Reply:
sure, updated